packages/@lwc/template-compiler/src/codegen/codegen.ts

```diff
@@ -612,65 +619,91 @@ export default class CodeGen {
     }
 
     genStaticParts(element: StaticElement): t.ArrayExpression | undefined {
-        const stack: (StaticElement | Text)[] = [element];
-        const partIdsToDatabagProps = new Map<number, t.Property[]>();
+        const stack: (StaticChildNode | Text[])[] = [element];
+        const partIdsToArgs = new Map<number, { text: t.Expression; databag: t.Expression }>();
         let partId = -1;
 
-        const addDatabagProp = (prop: t.Property) => {
-            let databags = partIdsToDatabagProps.get(partId);
-            if (!databags) {
-                databags = [];
-                partIdsToDatabagProps.set(partId, databags);
+        const getPartIdArgs = (partId: number) => {
+            let args = partIdsToArgs.get(partId);
+            if (!args) {
+                args = { text: t.literal(null), databag: t.literal(null) };
+                partIdsToArgs.set(partId, args);
             }
-            databags.push(prop);
+            return args;
+        };
+
+        const setPartIdText = (text: t.Expression) => {
+            const args = getPartIdArgs(partId)!;
+            args.text = text;
+        };
+
+        const setPartIdDatabag = (databag: t.Property[]) => {
+            const args = getPartIdArgs(partId)!;
+            args.databag = t.objectExpression(databag);
         };
 
         // Depth-first traversal. We assign a partId to each element, which is an integer based on traversal order.
         while (stack.length > 0) {
-            const node = stack.shift()!;
+            const current = stack.shift()!;
 
             // Skip comment nodes in parts count, as they will be stripped in production, unless when `lwc:preserve-comments` is enabled
-            if (!isComment(node) || this.preserveComments) {
+            if (isDynamicText(current) || !isComment(current) || this.preserveComments) {
                 partId++;
             }
 
-            if (isElement(node)) {
+            if (isDynamicText(current)) {
+                const textNodes = current;
+                const partToken = `${STATIC_PART_TOKEN_ID.TEXT}${partId}`;
+                // Use the first text node as the key.
+                // Dynamic text is guaranteed to have at least 1 text node in the array by transformStaticChildren.
+                this.staticExpressionMap.set(textNodes[0], partToken);
+                const concatenatedText = this.genConcatenatedText(
+                    textNodes.map(({ value }) =>
+                        isStringLiteral(value) ? value.value : this.bindExpression(value)
+                    )
+                );
+
+                setPartIdText(concatenatedText);
+                // addPartIdProp(t.property(t.identifier('text'), concatenatedText));
+            } else if (isElement(current)) {
+                const elm = current;
```

Comment:
This `elm` variable is unnecessary, correct?

Reply:
Yeah, I added it to help clarify what current is but it's not necessary.